data_structures/bloom_filters/bloom_filter/bloom_filter.h

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file bloom_filter.h
+ * @brief Declaration of a bloom filter
+ * https://en.wikipedia.org/wiki/Bloom_filter
+ * @author [Eric Breyer](https://github.com/ericbreyer)
+ * @see bloom_filter.c, main.c
+ */
+
+#include <stdio.h>   // logging
+#include <stdlib.h>  // free and calloc
+#include <string.h>  // memcpy
+#pragma once
+
+// Options for a set membership query.
+enum bloomResponse
+{
+    DEFINITELY_NOT_PRESENT = 0,
+    PROBABLY_PRESENT = 1
+};
+
+/**
+ * @brief A probabilistic set data structure supporting insertion
+ * that can make time and space efficient membership queries at the expense of
+ * a small probability of false positives (but no false negatives).
+ * In other words, a contains query can return either DEFINITELY_NOT_PRESENT
+ * or PROBABLY_PRESENT.
+ */
+struct BloomFilter;
+
+/**
+ * @brief Prints some stats about the filter.
+ * - Max Elements (n)
+ * - Hash Functions (k)
+ * - Slots (m)
+ * - Estimated Elements Inserted (n*)
+ * By nature of the filter we can not retrieve what elements were
+ * previously inserted, so this is the best version of a "to string" we
+ * can do
+ *
+ * @param this the bloomFilter to print the stats of
+ */
+void bloom_printStats(struct BloomFilter *this);
+
+/**
+ * @brief Inserts an item into the filter.
+ *
+ * @param[in] this the bloomFilter to insert an item into
+ * @param[in] elem the pointer to the element to insert
+ * @param[in] size the size of said elements in bytes
+ */
+void bloom_insert(struct BloomFilter *this, void *elem, size_t size);
+
+/**
+ * @brief Querys the filter for item membership an item into the filter.
```

Comment:
Wait, Is it just me or this sentence actually doesn't makes sense?
```suggestion
 * @brief Querys the filter for an item's membership in the filter.
```